picard/file.py

```diff
@@ -158,6 +158,21 @@ def _load(self, filename):
         """Load metadata from the file."""
         raise NotImplementedError
 
+    def _test_same(self):
+        if config.setting['clear_existing_tags']:
+            return False
+        new_metadata = self.new_metadata
+        orig_metadata = self.orig_metadata
+        tags = set(new_metadata.keys() + orig_metadata.keys())
+        for name in filter(lambda x: not x.startswith("~") and self.supports_tag(x), tags):
```

Comment:
`x.startswith("~")` is not a correct check. In some cases we do actually write `~` tags to files (I think `~rating` (which also depends on the email adress) and `~id3`-prefixed tags (look in `ID3File._save` for those)).